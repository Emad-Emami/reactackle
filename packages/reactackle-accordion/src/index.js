import Accordion from './Accordion';
import AccordionItem from './AccordionItem/AccordionItem';

export { Accordion, AccordionItem };
