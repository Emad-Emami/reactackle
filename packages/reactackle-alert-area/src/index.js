import AlertArea from './AlertArea';

export { AlertArea };
