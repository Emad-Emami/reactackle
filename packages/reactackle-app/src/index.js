import App from './App';

export { App };
export * from './BottomRegion';
export * from './TopRegion';
export * from './MainRegion';
