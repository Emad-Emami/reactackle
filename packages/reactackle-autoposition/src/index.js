import AutoPosition, { recalculateAllAutoPosition } from './AutoPosition';

export { AutoPosition, recalculateAllAutoPosition };
