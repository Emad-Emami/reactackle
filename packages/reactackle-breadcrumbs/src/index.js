import Breadcrumbs from './Breadcrumbs';

export { Breadcrumbs };
export * from './BreadcrumbsItem';
