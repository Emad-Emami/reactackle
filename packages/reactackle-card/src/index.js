import Card from './Card';

export { Card };
export * from './CardActions';
export * from './CardAreaMain';
export * from './CardAreaSide';
export * from './CardContent';
export * from './CardHeaderPrimary';
export * from './CardHeaderSecondary';
export * from './CardMedia';
