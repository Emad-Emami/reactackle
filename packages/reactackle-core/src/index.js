// theme
// defaultTheme
// composeReactackleTheme
// defaultThemeBlueprintBroadcast
// registerDefaultComponentTheme ++
// extractThemeOrDefault ++

// styleHelpers - for combining local theme
// export everything

// styleFunctions
// export everything

// - resetStyles
// - globalStyles
// - heladerClasses

// base
// export nothing

// constants
// export everything

// mixins
// export everything

// import {
//   registerDefaultComponentTheme,
//   extractThemeOrDefault,
// } from './theme';

// export {
//   registerDefaultComponentTheme,
//   extractThemeOrDefault,
// };

export * from './theme';
export * from './mixins';
export * from './styleFunctions';
export * from './styleHelpers';
export * from './utils';
export * from './constants/constants';
export * from './hoc/withExternalProps';
export * from './globalStyles';
export * from './resetStyles';
export * from './Theme/Theme';
export * from './Theme/withTheme';
