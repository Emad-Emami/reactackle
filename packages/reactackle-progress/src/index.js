import { registerDefaultComponentTheme } from 'reactackle-core';
import componentTheme from './styles/theme';

registerDefaultComponentTheme('progress', componentTheme);

export * from './CircleProgress/CircleProgress';
export * from './LinearProgress/LinearProgress';
export * from './Preloader/Preloader';
