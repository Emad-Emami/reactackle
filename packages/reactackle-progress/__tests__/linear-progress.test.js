import React from 'react';
import renderer from 'react-test-renderer';

import { LinearProgress } from '../src';

jest.mock('react-dom');

describe('<LinearProgress />', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <LinearProgress />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with ariaValueText props set to String', () => {
    const tree = renderer.create(
      <LinearProgress ariaValueText='Some aria value' />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with dynamicColor props set to true', () => {
    const tree = renderer.create(
      <LinearProgress dynamicColor />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with exactProgressLabel props set to true', () => {
    const tree = renderer.create(
      <LinearProgress exactProgressLabel />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with indeterminate props set to true', () => {
    const tree = renderer.create(
      <LinearProgress indeterminate />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with labelPositionX props set to one of enum value', () => {
    const tree = renderer.create(
      <LinearProgress labelPositionX='center' />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with labelPositionY props set to one of enum value', () => {
    const tree = renderer.create(
      <LinearProgress labelPositionY='center' />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with labelPostfix props set to String', () => {
    const tree = renderer.create(
      <LinearProgress labelPostfix='some string' />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with max props set to Number', () => {
    const tree = renderer.create(
      <LinearProgress max={50} />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with min props set to Number', () => {
    const tree = renderer.create(
      <LinearProgress min={20} />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with progressLabel props set to true', () => {
    const tree = renderer.create(
      <LinearProgress progressLabel />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with secondaryProgressValue props set to Number', () => {
    const tree = renderer.create(
      <LinearProgress secondaryProgressValue={2} />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with subtitle props set to String', () => {
    const tree = renderer.create(
      <LinearProgress subtitle='Some string' />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with value props set to Number', () => {
    const tree = renderer.create(
      <LinearProgress value={2} />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with withSecondaryProgress props set to true', () => {
    const tree = renderer.create(
      <LinearProgress withSecondaryProgress />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});