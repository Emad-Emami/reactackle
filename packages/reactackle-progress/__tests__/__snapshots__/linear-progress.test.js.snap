// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinearProgress /> renders correctly with ariaValueText props set to String 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="0"
  aria-valuetext="Some aria value"
  className="ProgressBoxStyled-ciJUOb enQWBS"
  role="progressbar"
>
  <div
    className="LinearProgressStyled-bYugBt eXJaSf"
  >
    <div
      className="ContentWrapperStyled-dHfVBA llocGx"
    >
      <div
        className="PrimaryLineStyled-brPmnZ ehYOcW"
        style={
          Object {
            "backgroundColor": "rgb(33, 150, 243)",
            "transform": "scaleX(0)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<LinearProgress /> renders correctly with default props 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="0"
  className="ProgressBoxStyled-ciJUOb enQWBS"
  role="progressbar"
>
  <div
    className="LinearProgressStyled-bYugBt eXJaSf"
  >
    <div
      className="ContentWrapperStyled-dHfVBA llocGx"
    >
      <div
        className="PrimaryLineStyled-brPmnZ ehYOcW"
        style={
          Object {
            "backgroundColor": "rgb(33, 150, 243)",
            "transform": "scaleX(0)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<LinearProgress /> renders correctly with dynamicColor props set to true 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="0"
  className="ProgressBoxStyled-ciJUOb enQWBS"
  role="progressbar"
>
  <div
    className="LinearProgressStyled-bYugBt eXJaSf"
  >
    <div
      className="ContentWrapperStyled-dHfVBA llocGx"
    >
      <div
        className="PrimaryLineStyled-brPmnZ ehYOcW"
        style={
          Object {
            "backgroundColor": "rgb(255, 195, 15)",
            "transform": "scaleX(0)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<LinearProgress /> renders correctly with exactProgressLabel props set to true 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="0"
  className="ProgressBoxStyled-ciJUOb enQWBS"
  role="progressbar"
>
  <div
    className="LinearProgressStyled-bYugBt eXJaSf"
  >
    <div
      className="LabelWrapperStyled-eKVrkZ cMhIgn"
    >
      <label
        className="ValueLabelStyled-Lgqql NdteF"
      >
        0
      </label>
    </div>
    <div
      className="ContentWrapperStyled-dHfVBA llocGx"
    >
      <div
        className="PrimaryLineStyled-brPmnZ ehYOcW"
        style={
          Object {
            "backgroundColor": "rgb(33, 150, 243)",
            "transform": "scaleX(0)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<LinearProgress /> renders correctly with indeterminate props set to true 1`] = `
<div
  className="ProgressBoxStyled-ciJUOb enQWBS"
  role="progressbar"
>
  <div
    className="LinearProgressStyled-bYugBt eXJaSf"
  >
    <div
      className="ContentWrapperStyled-dHfVBA llocGx"
    >
      <div
        className="PrimaryLineStyled-brPmnZ fuWMLn"
        style={
          Object {
            "backgroundColor": "rgb(33, 150, 243)",
          }
        }
      >
        <div
          className="SplitterStyled-jQZnrY pPBcv"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<LinearProgress /> renders correctly with labelPositionX props set to one of enum value 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="0"
  className="ProgressBoxStyled-ciJUOb enQWBS"
  role="progressbar"
>
  <div
    className="LinearProgressStyled-bYugBt eXJaSf"
  >
    <div
      className="ContentWrapperStyled-dHfVBA llocGx"
    >
      <div
        className="PrimaryLineStyled-brPmnZ ehYOcW"
        style={
          Object {
            "backgroundColor": "rgb(33, 150, 243)",
            "transform": "scaleX(0)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<LinearProgress /> renders correctly with labelPositionY props set to one of enum value 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="0"
  className="ProgressBoxStyled-ciJUOb enQWBS"
  role="progressbar"
>
  <div
    className="LinearProgressStyled-bYugBt eXJaSf"
  >
    <div
      className="ContentWrapperStyled-dHfVBA llocGx"
    >
      <div
        className="PrimaryLineStyled-brPmnZ ehYOcW"
        style={
          Object {
            "backgroundColor": "rgb(33, 150, 243)",
            "transform": "scaleX(0)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<LinearProgress /> renders correctly with labelPostfix props set to String 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="0"
  className="ProgressBoxStyled-ciJUOb enQWBS"
  role="progressbar"
>
  <div
    className="LinearProgressStyled-bYugBt eXJaSf"
  >
    <div
      className="ContentWrapperStyled-dHfVBA llocGx"
    >
      <div
        className="PrimaryLineStyled-brPmnZ ehYOcW"
        style={
          Object {
            "backgroundColor": "rgb(33, 150, 243)",
            "transform": "scaleX(0)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<LinearProgress /> renders correctly with max props set to Number 1`] = `
<div
  aria-valuemax="50"
  aria-valuemin="0"
  aria-valuenow="0"
  className="ProgressBoxStyled-ciJUOb enQWBS"
  role="progressbar"
>
  <div
    className="LinearProgressStyled-bYugBt eXJaSf"
  >
    <div
      className="ContentWrapperStyled-dHfVBA llocGx"
    >
      <div
        className="PrimaryLineStyled-brPmnZ ehYOcW"
        style={
          Object {
            "backgroundColor": "rgb(33, 150, 243)",
            "transform": "scaleX(0)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<LinearProgress /> renders correctly with min props set to Number 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="20"
  aria-valuenow="0"
  className="ProgressBoxStyled-ciJUOb enQWBS"
  role="progressbar"
>
  <div
    className="LinearProgressStyled-bYugBt eXJaSf"
  >
    <div
      className="ContentWrapperStyled-dHfVBA llocGx"
    >
      <div
        className="PrimaryLineStyled-brPmnZ ehYOcW"
        style={
          Object {
            "backgroundColor": "rgb(33, 150, 243)",
            "transform": "scaleX(0)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<LinearProgress /> renders correctly with progressLabel props set to true 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="0"
  className="ProgressBoxStyled-ciJUOb enQWBS"
  role="progressbar"
>
  <div
    className="LinearProgressStyled-bYugBt eXJaSf"
  >
    <div
      className="LabelWrapperStyled-eKVrkZ cMhIgn"
    >
      <label
        className="ValueLabelStyled-Lgqql NdteF"
      >
        0%
      </label>
    </div>
    <div
      className="ContentWrapperStyled-dHfVBA llocGx"
    >
      <div
        className="PrimaryLineStyled-brPmnZ ehYOcW"
        style={
          Object {
            "backgroundColor": "rgb(33, 150, 243)",
            "transform": "scaleX(0)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<LinearProgress /> renders correctly with secondaryProgressValue props set to Number 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="0"
  className="ProgressBoxStyled-ciJUOb enQWBS"
  role="progressbar"
>
  <div
    className="LinearProgressStyled-bYugBt eXJaSf"
  >
    <div
      className="ContentWrapperStyled-dHfVBA llocGx"
    >
      <div
        className="PrimaryLineStyled-brPmnZ ehYOcW"
        style={
          Object {
            "backgroundColor": "rgb(33, 150, 243)",
            "transform": "scaleX(0)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<LinearProgress /> renders correctly with subtitle props set to String 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="0"
  aria-valuetext="Some string"
  className="ProgressBoxStyled-ciJUOb enQWBS"
  role="progressbar"
>
  <div
    className="LinearProgressStyled-bYugBt eXJaSf"
  >
    <div
      className="LabelWrapperStyled-eKVrkZ cMhIgn"
    >
      <div
        className="SupplementTextStyled-gDaCZq NDsTy"
      >
        Some string
      </div>
    </div>
    <div
      className="ContentWrapperStyled-dHfVBA llocGx"
    >
      <div
        className="PrimaryLineStyled-brPmnZ ehYOcW"
        style={
          Object {
            "backgroundColor": "rgb(33, 150, 243)",
            "transform": "scaleX(0)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<LinearProgress /> renders correctly with value props set to Number 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="2"
  className="ProgressBoxStyled-ciJUOb enQWBS"
  role="progressbar"
>
  <div
    className="LinearProgressStyled-bYugBt eXJaSf"
  >
    <div
      className="ContentWrapperStyled-dHfVBA llocGx"
    >
      <div
        className="PrimaryLineStyled-brPmnZ ehYOcW"
        style={
          Object {
            "backgroundColor": "rgb(33, 150, 243)",
            "transform": "scaleX(0.02)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<LinearProgress /> renders correctly with withSecondaryProgress props set to true 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="0"
  className="ProgressBoxStyled-ciJUOb enQWBS"
  role="progressbar"
>
  <div
    className="LinearProgressStyled-bYugBt eXJaSf"
  >
    <div
      className="ContentWrapperStyled-dHfVBA llocGx"
    >
      <div
        className="SecondaryLineStyled-kfyfry eOWLqx"
        style={
          Object {
            "backgroundColor": "rgba(33, 150, 243, 0.15000000000000002)",
            "transform": "scaleX(0)",
          }
        }
      />
      <div
        className="PrimaryLineStyled-brPmnZ ehYOcW"
        style={
          Object {
            "backgroundColor": "rgb(33, 150, 243)",
            "transform": "scaleX(0)",
          }
        }
      />
    </div>
  </div>
</div>
`;
